src/libYARP_sig/src/Image.cpp

```diff
@@ -609,7 +609,40 @@ int ImageStorage::_pad_bytes (int linesize, int align) const
 
 
 
+std::map<YarpVocabPixelTypesEnum, unsigned int> Image::create_map()
+{
+    std::map<YarpVocabPixelTypesEnum, unsigned int> m;
+    m[VOCAB_PIXEL_INVALID]               = 0;
+    m[VOCAB_PIXEL_MONO]                  = 1;
+    m[VOCAB_PIXEL_MONO16]                = 2;
+    m[VOCAB_PIXEL_RGB]                   = 3;
+    m[VOCAB_PIXEL_RGBA]                  = 4;
+    m[VOCAB_PIXEL_BGRA]                  = 4;
+    m[VOCAB_PIXEL_INT]                   = 4;
+    m[VOCAB_PIXEL_HSV]                   = 3;
+    m[VOCAB_PIXEL_BGR]                   = 3;
+    m[VOCAB_PIXEL_MONO_SIGNED]           = 1;
+    m[VOCAB_PIXEL_RGB_SIGNED]            = 3;
+    m[VOCAB_PIXEL_RGB_INT]               = 4;
+    m[VOCAB_PIXEL_MONO_FLOAT]            = 4;
+    m[VOCAB_PIXEL_RGB_FLOAT]             = 12;
+    m[VOCAB_PIXEL_HSV_FLOAT]             = 12;
+    m[VOCAB_PIXEL_ENCODING_BAYER_GRBG8]  = 4;
+    m[VOCAB_PIXEL_ENCODING_BAYER_GRBG16] = 8;
+    m[VOCAB_PIXEL_ENCODING_BAYER_BGGR8]  = 4;
+    m[VOCAB_PIXEL_ENCODING_BAYER_BGGR16] = 8;
+    m[VOCAB_PIXEL_ENCODING_BAYER_GBRG8]  = 4;
+    m[VOCAB_PIXEL_ENCODING_BAYER_GBRG16] = 8;
+    m[VOCAB_PIXEL_ENCODING_BAYER_RGGB8]  = 4;
+    m[VOCAB_PIXEL_ENCODING_BAYER_RGGB16] = 8;
+    m[VOCAB_PIXEL_YUV_420]               = 3;
+    m[VOCAB_PIXEL_YUV_444]               = 3;
+    m[VOCAB_PIXEL_YUV_422]               = 3;
+    m[VOCAB_PIXEL_YUV_411]               = 3;
```

Comment:
please use `sizeof(<type>)` instead of constants, at least for the ones that correspond to a specific type.

Reply:
Also we are full c++11 now, you should be initializing it using c++11 syntax, i.e. something like this:

```c++
const std::map<YarpVocabPixelTypesEnum, unsigned int> Image::pixelCode2Size = {
   {VOCAB_PIXEL_INVALID, 0},
   {VOCAB_PIXEL_MONO,    sizeof(yarp::sig::PixelMono)},
   {VOCAB_PIXEL_MONO16,  sizeof(yarp::sig::PixelMono16)},
// ... 
};